proposals/0001-lhls.md

```diff
@@ -0,0 +1,313 @@
+
+#### WebRTC
+
+WebRTC relies on UDP for data delivery. UDP achieves significantly lower latency than TCP because it is lossy - it does not spend time guaranteeing the transmission of data. Lost data in a media stream is usually not fatal and results in dropped frames. WebRTC has been shown to achieve sub-second streaming latency.
+
+WebRTC suffers from scalability. Each "peer", in this case the client, requires a direct connection to another peer. Streaming servers are able to act as peers but must serve each request with a unique connection. This requirement precludes edge caching. A fitting analogy is that WebRTC streams require a unique URL for every client. WebRTC does have the advantage that peers may also be clients but this requires additional code.
+
+WebRTC also is less accessible than HTTP. Most developers today do not stream via WebRTC, and may not have the knowledge or resources to do so. Client support for WebRTC is available but more limited than existing HTTP clients. SSAI solutions are possible but also require further changes, and the breadth of support is not clear.
+
+## Method of Advertising LHLS Segments
+
+#### In-Playlist Signaling
+
+In-playlist signaling happens when prefetch segments are advertised within the same playlist as complete segments. Prefetch segments are included at the bottom and are differentiated via the `EXT-X-PREFETCH` tag. This is the scheme outlined in this proposal. 
+
+We chose this scheme because it is backwards compatible with clients which have not implemented this proposal. According to the HLS specification clients must ignore any tags not outlined within; since `EXT-X-PREFETCH` is not an official tag clients will be left with a standard HLS playlist. This allows a server to deliver a single manifest which plays in both LHLS and unsupported clients. This scheme also allows clients to disable or enable LHLS while the client is streaming. Developers may make this option available to the user so that they are able to choose an experience most beneficial to them.
+
+Another benefit of this scheme is that it more easily takes advantage of the HLS specification. In most cases an in-playlist prefetch segments can be treated like a complete segments, which promotes code reuse and eases implementation. As an example a prefetch segment's Media Sequence Number is calculated relative to the complete segments in the playlist; if prefetch segments were elsewhere a new solution would have to be devised.
+
+#### All-Playlist Signaling
+
+All-Playlist signaling is when every Media Playlist contains only prefetch segments. In this scheme the `EXT-X-PREFETCH` tag is not required; all segments are assumed to be prefetch. This solution has the benefit of requiring little or no modifications from the HLS specification (what changes, if any, are hard to determine without actually making such a proposal). But this solution is not backwards compatible with clients not supporting this proposal.
```

Comment:
```suggestion
All-Playlist signaling is a strategy in which every Media Playlist contains only prefetch segments. In this scheme, the `EXT-X-PREFETCH` tag is not required; all segments are assumed to be prefetch. This solution has the benefit of requiring little or no modifications from the HLS specification (what changes, if any, are hard to determine without actually making such a proposal). But this solution is not backwards compatible with clients not supporting this proposal.
```
It would be worth to mention the exact compatibility issue:
> Any Media Segment that is specified in a Playlist loaded by a client
> MUST be available for immediate download unless it has been marked
> with an EXT-X-GAP tag; otherwise playback errors can occur.  Once
> download starts, its transfer rate SHOULD NOT be constrained by the
> segment production process.

(Section 6.2.1 of the [Pantos RFC](https://tools.ietf.org/html/draft-pantos-hls-rfc8216bis-03))